R/query-string.R

```diff
@@ -22,13 +22,20 @@ parseQS <- function(qs){
   parts <- strsplit(qs, "&", fixed = TRUE)[[1]]
   kv <- strsplit(parts, "=", fixed = TRUE)
   kv <- kv[sapply(kv, length) == 2] # Ignore incompletes
+  if (length(kv) == 0L) return(list())
 
   keys <- sapply(kv, "[[", 1)
   keys <- unname(sapply(keys, URLdecode))
+  Encoding(keys) <- "UTF-8"
```

Comment:
@schloerke Perhaps this should give a warning instead if `keys` is non-ASCII. I believe that the encoding will be `unknown` iff the text contains only ASCII characters, but I'm not 100% sure.

Reply:
Yes it’s unknown for only ASCII strings but set it to UTF8 will cause no damage - it will stay in unknown.

Reply:
From my previous comment:

> we think that it's not even possible to call a function with argument names that can't be represented in the R session's native encoding

So if you have a function with a Chinese argument name and your machine is a Windows machine with a Chinese locale, you will not encounter this problem. You could run the same code on a Linux machine, and you won't have the problem. However, if you run the exact same code on a Windows machine with a non-Chinese locale, then you will end up with code that doesn't work, and error message that are very confusing.

My idea here is to warn people when they try to use non-ASCII characters so that they don't end up with that situation.